src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/TagHelperTooltipFactoryBase.cs

```diff
@@ -17,6 +20,57 @@ internal abstract class TagHelperTooltipFactoryBase
 
     private static readonly IReadOnlyList<char> s_newLineChars = new char[] { '\n', '\r' };
 
+    private readonly ISnapshotResolver _snapshotResolver;
+
+    protected TagHelperTooltipFactoryBase(ISnapshotResolver snapshotResolver)
+    {
+        _snapshotResolver = snapshotResolver;
+    }
+
+    internal string? GetProjectAvailability(string documentFilePath, string tagHelperTypeName)
+    {
+        if (!_snapshotResolver.TryResolveAllProjects(documentFilePath, out var projectSnapshots))
+        {
+            return null;
+        }
+
+        PooledStringBuilder? builder = null;
```

Comment:
Should we be using `Microsoft.VisualStudio.CoreUtility` in the language server? If the answer is no, maybe just use our `StringBuilderPool` instead rather than a different object pooling system?

Reply:
Whoops! I blame intellisense :P